ext/-test-/string/rb_interned_str.c

```diff
@@ -0,0 +1,14 @@
+#include "ruby.h"
+
+static VALUE
+bug_rb_interned_str_dup(VALUE self, VALUE str)
+{
+    rb_check_type(str, T_STRING);
```

Comment:
```suggestion
    Check_Type(str, T_STRING);
```
`Check_Type` is more common.